trieste/models/gpflow/sampler.py

```diff
@@ -795,24 +794,32 @@ def __init__(
         :param n_components: The desired number of features.
         """
 
+        super().__init__(model, n_components)
+
         if isinstance(model, SupportsGetInducingVariables):
-            inducing_points = model.get_inducing_variables()[0]  # [M, D]
+            self._inducing_points = model.get_inducing_variables()[0]  # [M, D]
         else:
-            inducing_points = model.get_internal_data().query_points  # [M, D]
+            self._inducing_points = model.get_internal_data().query_points  # [M, D]
 
-        self._cannonical_feature_functions = lambda x: tf.linalg.matrix_transpose(
-            model.get_kernel().K(inducing_points, x)
-        )
-
-        super().__init__(model, n_components)
+        if self.is_multioutput:
+            # mypy is confused and doesn't see that kernel can be of two types
+            kernel: MultioutputKernel = model.get_kernel()  # help mypy
+            self._canonical_feature_functions = lambda x: tf.linalg.matrix_transpose(
+                kernel.K(self._inducing_points, x, full_output_cov=False)
+            )
+        else:
+            kernel = model.get_kernel()
+            self._canonical_feature_functions = lambda x: tf.linalg.matrix_transpose(
+                kernel.K(self._inducing_points, x)
+            )
 
-    def __call__(self, x: TensorType) -> TensorType:  # [N,D] -> [N, L + M]
+    def call(self, x: TensorType) -> TensorType:  # [N, D] -> [N, F + M] or [L, N, F + M]
```

Comment:
What was the impact of this previously overriding and calling `__cal__l` rather than `call`? Were we skipping `FourierFeaturesBase.call` altogether?

Reply:
Not a significant impact for the current usage. `FourierFeaturesBase.call` was still being called because of the `super().__call__(x)` line. Keras `Layer.__call__` internally calls `build` (if needed) and `call`. One disadvantage of the old code would have been the dummy call in `__init__` to create the variables for the weights, instead of simply using `build`; but it turns out we don't need to create the variables at all.